Framework/PythonInterface/plugins/algorithms/ConvertWANDSCDtoQ.py

```diff
@@ -0,0 +1,319 @@
+
+        if normWS:
+            nd0 = normWS.getDimension(0)
+            nd1 = normWS.getDimension(1)
+            nd2 = normWS.getDimension(2)
+            if (nd0.name != d0.name or nd0.getNBins() != d0.getNBins()
+               or nd1.name != d1.name or nd1.getNBins() != d1.getNBins()
+               or nd2.name != d2.name):
+                issues["NormalisationWorkspace"] = "NormalisationWorkspace dimensions are not compatible with InputWorkspace"
+
+        ubWS = self.getProperty("UBWorkspace").value
+        if ubWS:
+            try:
+                sample = ubWS.sample()
+            except AttributeError:
+                sample = ubWS.getExperimentInfo(0).sample()
+            if not sample.hasOrientedLattice():
+                issues["UBWorkspace"] = "UBWorkspace does not has an OrientedLattice"
+        else:
+            if self.getProperty("Frame").value == 'HKL':
+                if not inWS.getExperimentInfo(0).sample().hasOrientedLattice():
+                    issues["Frame"] = "HKL selected but neither an UBWorkspace workspace was provided or " \
+                                      "the InputWorkspace has an OrientedLattice"
+
+        return issues
+
+    def PyExec(self):
+        inWS = self.getProperty("InputWorkspace").value
+        normWS = self.getProperty("NormalisationWorkspace").value
+        _norm = bool(normWS)
+
+        dim0_min, dim0_max, dim0_bins = self.getProperty('BinningDim0').value
+        dim1_min, dim1_max, dim1_bins = self.getProperty('BinningDim1').value
+        dim2_min, dim2_max, dim2_bins = self.getProperty('BinningDim2').value
+        dim0_bins = int(dim0_bins)
+        dim1_bins = int(dim1_bins)
+        dim2_bins = int(dim2_bins)
+        dim0_bin_size = (dim0_max-dim0_min)/dim0_bins
+        dim1_bin_size = (dim1_max-dim1_min)/dim1_bins
+        dim2_bin_size = (dim2_max-dim2_min)/dim2_bins
+
+        data_array = inWS.getSignalArray() # getSignalArray returns a F_CONTIGUOUS view of the signal array
+
+        number_of_runs = data_array.shape[2]
+
+        progress = Progress(self, 0.0, 1.0, number_of_runs+4)
+
+        # Get rotation array
+        s1 = np.deg2rad(inWS.getExperimentInfo(0).run().getProperty('s1').value)
+
+        normaliseBy = self.getProperty("NormaliseBy").value
+        if normaliseBy == "Monitor":
+            scale = np.asarray(inWS.getExperimentInfo(0).run().getProperty('monitor_count').value)
+        elif normaliseBy == "Time":
+            scale = np.asarray(inWS.getExperimentInfo(0).run().getProperty('duration').value)
+        else:
+            scale = np.ones(number_of_runs)
+
+        if _norm:
+            if normaliseBy == "Monitor":
+                norm_scale = np.sum(normWS.getExperimentInfo(0).run().getProperty('monitor_count').value)
+            elif normaliseBy == "Time":
+                norm_scale = np.sum(normWS.getExperimentInfo(0).run().getProperty('duration').value)
+            else:
+                norm_scale = 1.
```

Comment:
Just for consistency, this should be `norm_scale = float(number_of_runs)`

Reply:
I think this is fine, in either case it's not correctly normalised.